src/modules/company/index.ts

```diff
@@ -35,21 +35,24 @@ export class Company {
    * faker.company.name() // 'Zieme, Hauck and McClure'
    */
   name(format?: number): string {
-    const formats = [
-      '{{name.lastName}} {{company.companySuffix}}',
-      '{{name.lastName}} - {{name.lastName}}',
-      '{{name.lastName}}, {{name.lastName}} and {{name.lastName}}',
-    ];
-
     if (format != null) {
       deprecated({
         deprecated: 'faker.company.name(format)',
-        proposed: 'faker.company.name()',
+        proposed: 'faker.company.name() or faker.helpers.fake(format)',
         since: '7.4',
         until: '8.0',
       });
     }
 
+    // ToDo: This `staticFormats` pattern should be removed in the future. It is only used to maintain backwards compatibility.
+    const staticFormats = [
+      '{{name.lastName}} {{company.companySuffix}}',
+      '{{name.lastName}} - {{name.lastName}}',
+      '{{name.lastName}}, {{name.lastName}} and {{name.lastName}}',
+    ];
+    const localeFormats = this.faker.definitions.company.name_patterns;
+    const formats = localeFormats ? localeFormats : staticFormats;
```

Comment:
This can be simplified to 

```suggestion
    const formats = this.faker.definitions.company.name_patterns ?? staticFormats;
```

Reply:
It might be. I wasn't sure about the return when they don't have locales. But if it is `undefined` or `null` the `??` operator works fine. But if the return is `false` from locales the `??` operator doesn't work correctly.

Reply:
It will never be `false` as it's not typed as boolean, and it would fail then later on anyways, because we use it as array

And even then you could use `||` instead of `??` (but not needed here)